Controller/ContentManagement/CrudController.php

```diff
@@ -358,4 +374,47 @@ public function getContentTypeInfo(ContentType $contentType)
                 'contentType' => $contentType,
         ]);
     }
+
+    /**
+     * @Route("/api/user-profile", defaults={"_format": "json"}, methods={"GET"})
+     */
+    public function getUserProfile() : JsonResponse
+    {
+        $user = $this->getUser();
+        if (! $user instanceof User) {
+            throw new \RuntimeException('User profile class not recognized');
+        }
+        if (! $user->isEnabled()) {
+            throw new \RuntimeException('User disabled');
+        }
+
+        return $this->json([
+            'username' => $user->getUsername(),
+            'displayName' => $user->getDisplayName(),
+            'roles' => $user->getRoles(),
+            'email' => $user->getEmail(),
+            'circles' => $user->getCircles(),
+        ]);
+    }
+
+    /**
+     * @Route("/api/user-profiles", defaults={"_format": "json"}, methods={"GET"})
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function getUserProfiles() : JsonResponse
+    {
+        $users = [];
+        foreach ($this->userService->getAllUsers() as $user) {
+            if ($user->isEnabled()) {
+                $users[] = [
```

Comment:
aha the ->toArray() function can allready be re-used here!

Reply:
3d0a7d37515d4683568aae7b957bc3477a93cda7

Reply:
Done 9cbf75b7315e4fec4fa3edfeea82918a61565542

Reply:
even better in the user entity fda1219a36c1bec8f610426fdb8efc358057a658